File: app/src/main/java/com/example/village/home/search/SearchAdapter.java
package com.example.village.home.search;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.lifecycle.ViewModelStoreOwner;
import androidx.recyclerview.widget.RecyclerView;

import com.example.village.R;

import java.util.ArrayList;
import java.util.Objects;

public class SearchAdapter extends RecyclerView.Adapter<Holder> {

    SearchViewModel viewModel;

    SearchAdapter(Context context) {
        this.viewModel = new ViewModelProvider((ViewModelStoreOwner) context).get(SearchViewModel.class);
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        Context context = parent.getContext();
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = inflater.inflate(R.layout.searchword_item,parent,false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        Log.e("aa","position"+position+":  "+viewModel.searchWord.getValue().get(position));
        holder.text.setText(viewModel.searchWord.getValue().get(position));

        holder.delete_btn.setOnClickListener(v -> {
                removeItem(position);
        });

        holder.text.setOnClickListener(v -> {
            viewModel.setSearchEtvText(viewModel.searchWord.getValue().get(position));
        });

    }

    @Override
    public int getItemCount() {
        try {
            return viewModel.searchWord.getValue().size();
        } catch (NullPointerException e) {
            return 0;
        }
    }

    public void removeItem(int position) {
        if(position == 0 && viewModel.searchWord.getValue().size() == 0) {
            viewModel.searchWordDeleteIndex0.setValue(true);
            viewModel.emptyAlarm.setValue(true);
            viewModel.first = true;
        }
        viewModel.removeSearchWord(position);
        notifyItemRemoved(position);
        notifyDataSetChanged();

    }

}

class Holder extends RecyclerView.ViewHolder {
    TextView text;
    ImageButton delete_btn;

    public Holder(@NonNull View itemView) {
        super(itemView);
        this.text = itemView.findViewById(R.id.text);
        this.delete_btn = itemView.findViewById(R.id.delete_btn);
    }

}

File: app/src/main/java/com/example/village/home/search/SearchAdapter.kt
package com.example.village.home.search

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.village.databinding.SearchwordItemBinding

class SearchAdapter (private val context: Context) : RecyclerView.Adapter<SearchAdapter.SearchViewHolder>() {
    var data = listOf<SearchWord>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchAdapter.SearchViewHolder {
        val binding = SearchwordItemBinding.inflate(
                LayoutInflater.from(context), parent,false)

        return SearchViewHolder(binding)

    }

    override fun getItemCount(): Int = data.size

    override fun onBindViewHolder(holder: SearchAdapter.SearchViewHolder, position: Int, payloads: MutableList<Any>) {
        holder.onBind(data[position])
    }

    class SearchViewHolder(val binding : SearchwordItemBinding) : RecyclerView.ViewHolder(binding.root) {
        fun onBind(data : SearchWord) {
            binding.wordDataClass = data
        }
    }

    /**
     * Called by RecyclerView to display the data at the specified position. This method should
     * update the contents of the [ViewHolder.itemView] to reflect the item at the given
     * position.
     *
     *
     * Note that unlike [android.widget.ListView], RecyclerView will not call this method
     * again if the position of the item changes in the data set unless the item itself is
     * invalidated or the new position cannot be determined. For this reason, you should only
     * use the `position` parameter while acquiring the related data item inside
     * this method and should not keep a copy of it. If you need the position of an item later
     * on (e.g. in a click listener), use [ViewHolder.getAdapterPosition] which will
     * have the updated adapter position.
     *
     * Override [.onBindViewHolder] instead if Adapter can
     * handle efficient partial bind.
     *
     * @param holder The ViewHolder which should be updated to represent the contents of the
     * item at the given position in the data set.
     * @param position The position of the item within the adapter's data set.
     */
    override fun onBindViewHolder(holder: SearchViewHolder, position: Int) {
        TODO("Not yet implemented")
    }
}

